Update of Phase2 Tracker Digitizer to fix an issue in Premixing

suchandradutta opened the pull request:
#### PR description:
Updating Phase2TrackerDigitizerAlgorithm to fix the issue observed during Premixing of PU events at the Digitization. It was observed that the Digitized signal gets lowered and correspondingly #  Digis are lower than expected in inner Pixel layers . It was found that this is due to the fact that the Dual-Slope signal scaling is used. It is now fixed using simple liner scaling of charge during the first step  of Pre-Mixing.  

  - Changes are NOT expected! As shown in the validation plots, the PreMixing and noPreMixing show similar behaviour as expected.  
  - NO other PRs or externals it depends upon it
  - The details of the code change and the validation results are reported to Tk Phase2 Simulation group [Link](https://indico.cern.ch/event/1321219/contributions/5602295/attachments/2726551/4738571/Phase2TrackerDigi_PreMixing_Validation_29092023.pdf)
  
@emiglior

Commit:
updating Phase2TrackerDigitizerAlgorithm to fix the issue of lower charge in # of Digis in inner Pixel modules when Premixing is used during Digitization together with Dual Slope signal scaling

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42938/37085

- This PR adds an extra 24KB to repository

cmsbuild commented:
A new Pull Request was created by  @suchandradutta (Suchandra Dutta) for master.

It involves the following packages:

- SimTracker/SiPhase2Digitizer (**upgrade, simulation**)


@AdrianoDee, @cmsbuild, @mdhildreth, @civanch, @srimanob can you please review it and eventually sign? Thanks.
@dgulhan, @rovere, @VourMa, @mmusich, @missirol, @threus, @mtosi, @VinInn, @JanFSchulte, @GiacomoSguazzoni this is something you requested to watch as well.
@sextonkennedy, @rappoccio, @antoniovilela you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

srimanob commented:
test parameters:
- workflow = 25034.999

srimanob commented:
@cmsbuild please test

mmusich commented on SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:
```diff
@@ -233,11 +233,17 @@
         AddThresholdSmearing = False,
+        Phase2ReadoutMode = -1,
+        AdcFullScale = 255,
+        AddXTalk = False,
```
I would recommend documenting this here:
https://github.com/cms-sw/cmssw/blob/f738feebb2f2a520baac4e21cd3da20e3e52236e/SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py#L219-L223

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e80085/35029/summary.html
**COMMIT**: 12a00660025edcaa04b450f2bc5ddf5b84e65312
**CMSSW**: CMSSW_13_3_X_2023-10-03-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/42938/35029/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially added 3 lines to the logs
 - Reco comparison results: 2460 differences found in the comparisons
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3358320
 - DQMHistoTests: Total failures: 25646
 - DQMHistoTests: Total nulls: 35
 - DQMHistoTests: Total successes: 3332617
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 214 log files, 167 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/triggerResults): found differences in 1 / 48 workflows

srimanob commented:
Hi @suchandradutta 
Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).

I assume the change is expected, only Phase-2 PreMix is changed with this PR.

suchandradutta commented:
> Hi @suchandradutta Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).
> 
> I assume the change is expected, only Phase-2 PreMix is changed with this PR.

@srimanob can you please let me know the workflow number where the difference is seen? So that I can re-run and check.

srimanob commented:
> > Hi @suchandradutta Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).
> > I assume the change is expected, only Phase-2 PreMix is changed with this PR.
> 
> @srimanob can you please let me know the workflow number where the difference is seen? So that I can re-run and check.

25034.999

Commit:
updating the configuration file adding proper comments and switching off X-Talk for OT layers alsoat the first step of PreMixing

suchandradutta commented:
> Hi @suchandradutta Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).
> 
> I assume the change is expected, only Phase-2 PreMix is changed with this PR.

@srimanob I can fully confirm the changes observed in Inner Tracker Digi Occupancy/Charge histograms exactly as expected. I have updated the configuration file with some comments and switched off x-talk simulation also fo outer tracker layers in first step of PreMixing. Test were also done with the updated configuration file.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42938/37102

- This PR adds an extra 24KB to repository

srimanob commented:
@cmsbuild please test

civanch commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e80085/35092/summary.html
**COMMIT**: 8f397b73fa74f67d010102c3dfcd9b193e60b74e
**CMSSW**: CMSSW_13_3_X_2023-10-08-0000/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42938/35092/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially removed 1 lines from the logs
 - Reco comparison results: 3060 differences found in the comparisons
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3356920
 - DQMHistoTests: Total failures: 2019
 - DQMHistoTests: Total nulls: 35
 - DQMHistoTests: Total successes: 3354844
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 214 log files, 167 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-08-0000+e80085/59165/triggerResults): found differences in 1 / 48 workflows

civanch commented:
+1

only phase2 WFs are affected, which is expected

srimanob commented:
+Upgrade

Change on Phase-2 premix workflow is expected.
@civanch do you want the update to present at FullSim meeting?

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @rappoccio, @antoniovilela, @sextonkennedy (and backports should be raised in the release meeting by the corresponding L2)

civanch commented:
yes, we may discuss this Friday.

antoniovilela commented:
+1
